### DISCUSSION TITLE
Get file content from media item

### DISCUSSION THREAD

**Discussion by dragermrb** [spatie/laravel-medialibrary#2946] [Q&A] [answered]
Is there any way to get file content from media item?

I need to get file content from a media item, but have problems dealing with different disks. For example:

```php
$media = \Spatie\MediaLibrary\Models\Media::first();
$file_content = Storage::disk($media->disk)->get($media->getPath()); // Will work on S3 disk, fails on local disk
```

On S3 disk, `$media->getPath()` will return a relative path, but on local disk will returns a full path.

Any way to get file content no matter disk type?

**Comment by PaolaRuby** [comment]
Use a custom macro for choose wich path use, or use a custom path generator for make relative paths on local disk

**Comment by dragermrb** [comment]
Does a PR that implements `$media->getContent()` make sense?

**Reply by PaolaRuby** [reply]
Maybe, you could try

**Comment by Norris1z** [comment, marked as answer]
I needed this and this is how I did it:

```
$stream = $media->stream();
$content = stream_get_contents($stream);
```

**Comment by dragermrb** [comment]
Hi @Norris1z 

I have used something like this:

```php
function getMediaContent(Media $media)
{
    return config("filesystems.disks.{$media->disk}.driver") === 'local'
        ? file_get_contents($media->getPath())
        : Storage::disk($media->disk)->get($media->getPath());
}
```

But your answer seems beeter!
